atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -215,45 +192,122 @@ def _send_mail(self, receivers_list, subject, body):
 
         s = None
         try:
-            s = smtplib.SMTP(self.smtp_uri)
+            s = smtplib.SMTP(self.smtp_host)
             s.sendmail(self.from_address, receivers_list, msg.as_string())
         except (socket.gaierror, smtplib.SMTPException) as e:
-            raise PluginFailedException('Error communicating with SMTP server: %s' % str(e))
+            self.log.error('Error communicating with SMTP server')
+            raise e
         finally:
             if s is not None:
                 s.quit()
 
+    def _get_email_from_koji_obj(self, obj):
+        if obj.get('krb_principal'):
+            return obj['krb_principal'].lower()
+        else:
+            if not self.email_domain:
+                raise RuntimeError("Empty email_domain specified")
+            return '@'.join([obj['name'], self.email_domain])
+
+    def _get_koji_submitter(self):
+        koji_task_info = self.session.getTaskInfo(self.koji_task_id)
+        koji_task_owner = self.session.getUser(koji_task_info['owner'])
+        koji_task_owner_email = self._get_email_from_koji_obj(koji_task_owner).strip()
+        self.submitter = koji_task_owner_email
+        return koji_task_owner_email
+
+    def _get_koji_owners(self):
+        result = []
+        koji_build_info = self.session.getBuild(self.koji_build_id)
+
+        koji_tags = self.session.listTags(self.koji_build_id)
+        for koji_tag in koji_tags:
+            koji_tag_id = koji_tag['id']
+            koji_package_id = koji_build_info['package_id']
+            koji_pkg_tag_config = self.session.getPackageConfig(koji_tag_id, koji_package_id)
+            koji_pkg_tag_owner = self.session.getUser(koji_pkg_tag_config['owner_id'])
+
+            result.append(self._get_email_from_koji_obj(koji_pkg_tag_owner).strip())
+
+        return result
+
+    def _get_logs_url(self):
+        url = None
+        try:
+            # We're importing this here in order to trap ImportError
+            from koji import PathInfo
+            pathinfo = PathInfo(topdir=self.koji_root)
+            url = urljoin(pathinfo.work(), pathinfo.taskrelpath(self.koji_task_id))
+        except Exception:
+            self.log.exception("Failed to fetch logs from koji")
+            if self.url and self.workflow.openshift_build_selflink:
+                url = urljoin(self.url, self.workflow.openshift_build_selflink + '/log')
+        return url
+
+    def _get_receivers_list(self):
+        receivers_list = []
+        if self.additional_addresses:
+            receivers_list += [x.strip() for x in self.additional_addresses]
```

Comment:
Slight improvement might be to perform this stripping on `receivers_list` once it's been assembled, rather than in several different places before assembly.